File: hiretech_frontend/core/domain/interview.ts
export type Difficulty = "foundation" | "intermediate" | "advanced";
export type InterviewStatus = "draft" | "ready" | "invited" | "scheduled" | "in_progress" | "completed" | "cancelled" | "expired";

export interface TechnicalSkill {
  id: string;
  name: string;
  selected: boolean;
}

export interface Question {
  id: string;
  sequence: number;
  title: string;
  prompt: string;
  skill: string;
  difficulty: Difficulty;
  expectedMinutes: number;
}

export interface CandidateAnswer {
  id: string;
  questionId: string;
  response: string;
  code: string;
  language: string;
  savedAt: string;
}

export interface Interview {
  id: string;
  title: string;
  organizationId: string;
  status: InterviewStatus;
  scheduledAt: string;
  durationMinutes: number | null;
  difficulty: Difficulty | null;
  skills: TechnicalSkill[];
  candidateAlias: string;
  progress: number;
  expiresAt?: string;
  startedAt?: string | null;
  completedAt?: string | null;
}

export type ManagedQuestionType = "technical_discussion" | "coding" | "system_design" | "debugging";

export interface ManagedQuestion {
  id: string;
  interviewId: string;
  sequence: number;
  type: ManagedQuestionType;
  prompt: string;
  competencyIds: string[];
  difficulty: number;
  timeLimitSeconds: number | null;
  createdAt: string;
}

export interface RecruiterInterviewDetail extends Interview {
  candidateEmail: string;
  positionTitle: string;
  seniority: string;
  language: "tr" | "en";
  questionSource: "human" | "ai";
  rubricVersion: string;
  expiresAt: string;
  questions: ManagedQuestion[];
}

export interface AddManagedQuestionInput {
  interviewId: string;
  type: ManagedQuestionType;
  prompt: string;
  competencyIds: string[];
  difficulty: number;
  timeLimitSeconds: number | null;
}

export interface RequestQuestionDraftInput extends Omit<AddManagedQuestionInput, "prompt"> {
  taskBrief: string;
}

export interface ManagedQuestionDraft {
  id: string;
  interviewId: string;
  requestedBy: string;
  reviewedBy: string | null;
  type: ManagedQuestionType;
  prompt: string;
  competencyIds: string[];
  difficulty: number;
  timeLimitSeconds: number | null;
  language: "tr" | "en";
  status: "pending" | "approved" | "rejected";
  modelId: string;
  modelVersion: string;
  reviewNotes: string;
  createdAt: string;
  reviewedAt: string | null;
  canReview: boolean;
}

export interface InterviewInvitation {
  invitationId: string;
  token: string;
  expiresAt: string;
}

export interface CandidateWorkspace {
  interview: Interview;
  upcoming: Interview[];
  completedCount: number;
  averageFeedbackDelayHours: number;
}

export interface InterviewWorkspace {
  interview: Interview;
  questions: Question[];
  answers: CandidateAnswer[];
  activeQuestionId: string;
  remainingSeconds: number;
  deadline?: string;
}

export interface CandidateAnswerSubmission {
  interviewId: string;
  questionId: string;
  text: string;
  codeLanguage: string;
  codeContent: string;
  idempotencyKey: string;
  supersedesId?: string;
}

File: hiretech_backend/internal/domain/interview/model/interview.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type Status string

const (
	StatusDraft      Status = "draft"
	StatusReady      Status = "ready"
	StatusInvited    Status = "invited"
	StatusInProgress Status = "in_progress"
	StatusCompleted  Status = "completed"
	StatusCancelled  Status = "cancelled"
	StatusExpired    Status = "expired"
)

type InterviewLanguage string

const (
	LanguageTurkish InterviewLanguage = "tr"
	LanguageEnglish InterviewLanguage = "en"
)

type QuestionSource string

const (
	QuestionSourceHuman QuestionSource = "human"
	QuestionSourceAI    QuestionSource = "ai"
)

type Mode string

const (
	ModeAIDisabled      Mode = "ai_disabled"
	ModeGuidedAI        Mode = "guided_ai"
	ModeAICollaboration Mode = "ai_collaboration"
)

type Interview struct {
	ID                   uuid.UUID
	OrganizationID       uuid.UUID
	CreatedBy            uuid.UUID
	CandidateUserID      uuid.UUID
	CandidateEmail       string
	CandidateDisplayName string
	Title                string
	PositionTitle        string
	Seniority            string
	TechnologyTags       []string
	Mode                 Mode
	Language             InterviewLanguage
	QuestionSource       QuestionSource
	RubricVersion        string
	Status               Status
	StartsAt             *time.Time
	ExpiresAt            time.Time
	StartedAt            *time.Time
	CompletedAt          *time.Time
	CancelledAt          *time.Time
	Version              int
	CreatedAt            time.Time
	UpdatedAt            time.Time
	Questions            []*Question
	Answers              []*Answer
}

func (i *Interview) CanAddQuestion() bool {
	return i != nil && (i.Status == StatusDraft || i.Status == StatusReady)
}
func (i *Interview) CanPublish() bool {
	return i != nil && (i.Status == StatusReady || i.Status == StatusInvited)
}
func (i *Interview) IsTerminal() bool {
	return i == nil || i.Status == StatusCompleted || i.Status == StatusCancelled || i.Status == StatusExpired
}

type QuestionType string

const (
	QuestionTechnicalDiscussion QuestionType = "technical_discussion"
	QuestionCoding              QuestionType = "coding"
	QuestionSystemDesign        QuestionType = "system_design"
	QuestionDebugging           QuestionType = "debugging"
)

type Question struct {
	ID               uuid.UUID
	OrganizationID   uuid.UUID
	InterviewID      uuid.UUID
	Sequence         int
	Type             QuestionType
	Prompt           string
	CompetencyIDs    []string
	Difficulty       int
	TimeLimitSeconds *int
	CreatedAt        time.Time
}

type Invitation struct {
	ID             uuid.UUID
	OrganizationID uuid.UUID
	InterviewID    uuid.UUID
	TokenHash      string
	ExpiresAt      time.Time
	UsedAt         *time.Time
	RevokedAt      *time.Time
	CreatedAt      time.Time
}

type Consent struct {
	ID             uuid.UUID
	OrganizationID uuid.UUID
	InterviewID    uuid.UUID
	UserID         uuid.UUID
	PolicyVersion  string
	Locale         string
	Purpose        string
	AcceptedAt     time.Time
}

type SessionStatus string

const (
	SessionReady     SessionStatus = "ready"
	SessionActive    SessionStatus = "active"
	SessionCompleted SessionStatus = "completed"
	SessionExpired   SessionStatus = "expired"
)

type Session struct {
	ID             uuid.UUID
	OrganizationID uuid.UUID
	InterviewID    uuid.UUID
	UserID         uuid.UUID
	DeviceID       uuid.UUID
	Status         SessionStatus
	StartedAt      *time.Time
	CompletedAt    *time.Time
	LastSeenAt     time.Time
	Version        int
	CreatedAt      time.Time
}

type AnswerStatus string

const (
	AnswerSubmitted  AnswerStatus = "submitted"
	AnswerSuperseded AnswerStatus = "superseded"
)

type Answer struct {
	ID             uuid.UUID
	OrganizationID uuid.UUID
	InterviewID    uuid.UUID
	QuestionID     uuid.UUID
	SessionID      uuid.UUID
	UserID         uuid.UUID
	Status         AnswerStatus
	Text           *string
	CodeLanguage   *string
	CodeContent    *string
	ContentHash    string
	IdempotencyKey uuid.UUID
	SupersedesID   *uuid.UUID
	SubmittedAt    time.Time
	CreatedAt      time.Time
}

type AuditEvent struct {
	ID             uuid.UUID
	OrganizationID uuid.UUID
	ActorID        uuid.UUID
	SessionID      string
	DeviceID       uuid.UUID
	RequestID      string
	Action         string
	ResourceType   string
	ResourceID     uuid.UUID
	ParentID       uuid.UUID
	FromStatus     string
	ToStatus       string
	OccurredAt     time.Time
}
